src/core/steam_turbine.jl

```diff
@@ -0,0 +1,214 @@
+            else
+                @warn("Steam turbine inlet temperature will be calculated from inlet pressure and specified superheat")
+            end
+        end
+    end
+
+    if isnan(st.electric_produced_to_thermal_consumed_ratio) || isnan(thermal_produced_to_thermal_consumed_ratio)
+        assign_st_elec_and_therm_prod_ratios!(st)
+    end
+
+    return st
+end
+
+
+"""
+    get_steam_turbine_defaults(size_class::Int)
+
+return a Dict{String, Float64} by selecting the appropriate values from 
+data/steam_turbine/steam_turbine_default_data.json, which contains values based on size_class for the 
+custom_st_inputs, i.e.
+- "installed_cost_per_kw"
+- "om_cost_per_kwh"
+- "inlet_steam_pressure_psig"
+- "inlet_steam_temperature_degF"
+- "outlet_steam_pressure_psig",
+- "isentropic_efficiency"
+- "gearbox_generator_efficiency"
+- "net_to_gross_electric_ratio"
+"""
+function get_steam_turbine_defaults(size_class::Int)
+    defaults = JSON.parsefile(joinpath(dirname(@__FILE__), "..", "..", "data", "steam_turbine", "steam_turbine_default_data.json"))
+    steam_turbine_defaults = Dict{String, Any}()
+
+    for key in keys(defaults)
+        steam_turbine_defaults[key] = defaults[key][size_class]
+    end
+    defaults = nothing
+
+    return steam_turbine_defaults
+end
+
+"""
+    assign_st_elec_and_therm_prod_ratios!(st::SteamTurbine) 
+
+Calculate steam turbine (ST) electric output to thermal input ratio based on inlet and outlet steam conditions and ST performance.
+This function uses the CoolProp package to calculate steam properties.
+    Units of [kWe_net / kWt_in]
+:return: st_elec_out_to_therm_in_ratio, st_therm_out_to_therm_in_ratio
+
+"""
+function assign_st_elec_and_therm_prod_ratios!(st::SteamTurbine)
+
+
+    # Convert input steam conditions to SI (absolute pressures, not gauge)
+    # ST Inlet
+    p_in_pa = (st.inlet_steam_pressure_psig / 14.5038 + 1.01325) * 1.0E5
+    if isnan(st.inlet_steam_temperature_degF)
+        t_in_sat_k = PropsSI("T","P",p_in_pa,"Q",1.0,"Water")
+        t_superheat_in_k = (st.inlet_steam_superheat_degF - 32.0) * 5.0 / 9.0 + 273.15
+        t_in_k = t_in_sat_k + t_superheat_in_k
+    else
+        t_in_k = (st.inlet_steam_temperature_degF - 32.0) * 5.0 / 9.0 + 273.15
+    end
+    h_in_j_per_kg = PropsSI("H","P",p_in_pa,"T",t_in_k,"Water")
+    s_in_j_per_kgK = PropsSI("S","P",p_in_pa,"T",t_in_k,"Water")
+
+    # ST Outlet
+    p_out_pa = (st.outlet_steam_pressure_psig / 14.5038 + 1.01325) * 1.0E5
+    h_out_ideal_j_per_kg = PropsSI("H","P",p_out_pa,"S",s_in_j_per_kgK,"Water")
+    h_out_j_per_kg = h_in_j_per_kg - st.isentropic_efficiency * (h_in_j_per_kg - h_out_ideal_j_per_kg)
+    x_out = PropsSI("Q","P",p_out_pa,"H",h_out_j_per_kg,"Water")
+
+    if x_out != -1.0 && x_out < st.outlet_steam_min_vapor_fraction
+        error("The calculated steam outlet vapor fraction of $x_out is lower than the minimum allowable value of $(st.outlet_steam_min_vapor_fraction)")
+    end
```

Comment:
these compare well to the python code, but is there source material for this somewhere that we can cite?

Reply:
I agree we should document this better at least, especially the reason for the limits on steam outlet vapor quality, and the (later/down farther) factors for the electrical generator efficiency and the other/auxiliary (e.g. pumps) loss factor. Otherwise, these are basic isentropic efficiency calculations out of an undergrad thermo text.

Reply:
I took a stab at adding some descriptions to non-obvious parameters names, and added some more descriptive notes above each section of the isentropic efficiency calcs in 8647ca1315b10a7e5c8558e11636acb3963561c3.